### PR TITLE
SLE-511 Subscribe for server events

### PR DESCRIPTION
**Pull request by damien-urruty-sonarsource**

### CODE REVIEW

File: its/build.properties [hunk 1]
```diff
@@ -1,4 +1,6 @@
 source.. = src/
-output.. = target/classes/
+output.. = target/classes/,\
+           bin/
```
Comment (henryju) [inline its/build.properties:3]: Are you sure this change is needed?

File: its/src/org/sonarlint/eclipse/its/SonarQubeConnectedModeTest.java [hunk 2]
```diff
@@ -174,10 +206,55 @@ public void testLocalServerStatusRequest() throws Exception {
...
+    assertThat(defaultEditor.getMarkers())
+      .extracting(Marker::getText, Marker::getLineNumber)
+      .containsOnly(
+        tuple("Replace this use of System.out or System.err by a logger.", 9));
+
+    var qualityProfile = getQualityProfile(JAVA_SIMPLE_PROJECT_KEY, "SonarLint IT Java");
+    deactivateRule(qualityProfile, "S106");
+    Thread.sleep(5000);
+
+    doAndWaitForSonarLintAnalysisJob(() -> {
```
Comment (henryju) [inline its/src/org/sonarlint/eclipse/its/SonarQubeConnectedModeTest.java:249]: Would be great at some point to have analysis automatically triggered when rule is changed.

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/engine/connected/ConnectedEngineFacade.java [hunk 3]
```diff
@@ -364,6 +367,17 @@ public void updateProjectList(IProgressMonitor monitor) {
     });
   }
 
+  @Override
+  public Set<String> getBoundProjectKeys() {
+    return ProjectsProviderUtils.allProjects().stream()
```
Comment (henryju) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/engine/connected/ConnectedEngineFacade.java:372]: Maybe you can share part of this code with the following method `getBoundProjects`?

**Review by henryju — APPROVED** [review]

File: its/src/org/sonarlint/eclipse/its/SonarQubeConnectedModeTest.java (hunk 2 above)
Comment (damien-urruty-sonarsource) [reply its/src/org/sonarlint/eclipse/its/SonarQubeConnectedModeTest.java:249]: We haven't done it on SLI either, we could consider it later yes

**Pull request merged**